### PR TITLE
[libresources] Add new port

### PR DESCRIPTION
**Pull request by as-shura**

### CODE REVIEW

**Comment by as-shura** [comment]
Can someone explain to me what I need to do to resolve the issues. Is it from my side or not?

**Comment by LilyWangLL** [comment]
This port build failed on windows, you can download the failure log from: https://dev.azure.com/vcpkg/public/_build/results?buildId=85131&view=artifacts&pathAsName=false&type=publishedArtifacts.

**Comment by as-shura** [comment]
> This port build failed on windows, you can download the failure log from: https://dev.azure.com/vcpkg/public/_build/results?buildId=85131&view=artifacts&pathAsName=false&type=publishedArtifacts.

Weird although I do the check for 32/64 bits on 32 bits build the log shows:

```
[1/2] "D:/downloads/tools/cmake-3.25.1-windows/cmake-3.25.1-windows-i386/bin/cmake.exe" -E chdir "../../x86-windows-dbg" "D:/downloads/tools/cmake-3.25.1-windows/cmake-3.25.1-windows-i386/bin/cmake.exe" "D:/buildtrees/libresources/src/f1db3b80b7-f2c6d5164c.clean" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/packages/libresources_x86-windows/debug" "-DFETCHCONTENT_FULLY_DISCONNECTED=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files/Microsoft Visual Studio/2022/Enterprise/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=OFF" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=C:/a/1/s/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v143" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=C:/a/1/s/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_LINKER_FLAGS_RELEASE=" "-DVCPKG_LINKER_FLAGS_DEBUG=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-D_VCPKG_ROOT_DIR=C:/a/1/s" "-D_VCPKG_INSTALLED_DIR=D:/installed" "-DVCPKG_MANIFEST_INSTALL=OFF"
-- The C compiler identification is MSVC 19.34.31937.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Enterprise/VC/Tools/MSVC/14.34.31933/bin/Hostx64/x86/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
bin2coff_EXECUTABLE = D:/buildtrees/libresources/x86-windows-dbg/Utilities/bin2coff/bin2coff.exe
ARCHITECTURE : x64
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_INSTALL_BINDIR
    FETCHCONTENT_FULLY_DISCONNECTED
```

**Comment by as-shura** [comment]
You can see here : https://github.com/as-shura/libresources/blob/master/additional_execution.cmake

**Comment by as-shura** [comment]
Merge please

**Comment by dg0yt** [comment]
It cannot be merged. You have to remove the unrelated mupdf changes.
And maybe the name is too generic,
Just wait for proper review.

File: ports/libmupdf/portfile.cmake [hunk 1]
```diff
@@ -21,9 +20,14 @@ vcpkg_cmake_configure(
...
+vcpkg_cmake_config_fixup(PACKAGE_NAME libmupdf CONFIG_PATH lib/cmake/libmupdf)
 
 file(COPY "${SOURCE_PATH}/include/mupdf" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
 
 vcpkg_copy_pdbs()
 
 file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+
+
+
```
Comment (LilyWangLL) [inline ports/libmupdf/portfile.cmake:33]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/COPYING")

```

File: ports/libresources/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "libresources",
+  "version": "1.0.0",
+  "port-version": 1,
```
Comment (LilyWangLL) [inline ports/libresources/vcpkg.json:4]:
```suggestion
```

File: ports/libresources/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+vcpkg_cmake_config_fixup(PACKAGE_NAME libresources CONFIG_PATH lib/cmake/libresources)
+
+vcpkg_copy_pdbs()
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+file(INSTALL "${SOURCE_PATH}/COPYING" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+
+
+
```
Comment (LilyWangLL) [inline ports/libresources/portfile.cmake:26]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/COPYING")

```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
About `libmupdf/CMakeLists.txt`, please refer to PR #29635.

File: ports/libmupdf/portfile.cmake (hunk 1 above)
Comment (as-shura) [reply ports/libmupdf/portfile.cmake]: libmupdf was included by error.

Comment (as-shura) [reply ports/libmupdf/portfile.cmake]: As stated by https://github.com/microsoft/vcpkg/pull/29563#issuecomment-1429100755

Comment (as-shura) [reply ports/libmupdf/portfile.cmake]: This is libresources and not libmupdf

**Comment by dg0yt** [comment]
We didn't want the mupdf *files* to be removed but the mupdf *changes*.

**Comment by as-shura** [comment]
I don't understandI think I will close this PR.

**Comment by LilyWangLL** [comment]
Closing this PR since it seems that no progress is being made. Please ping us to reopen if work is still being done.

**Pull request closed without merging**